src/stats/boxplot.jl

```diff
@@ -95,7 +95,7 @@ function Makie.plot!(plot::BoxPlot)
         notchmax = Float32[]
         t_segments = Point2f0[]
         outlier_indices = Int[]
-        boxcolor = []
+        boxcolor = eltype(plot[:color].val)[]
```

Comment:
I'd use RGBAf0, which is what all colors get converted to internally, and then when pushing to it use `push!(boxcolor, to_color(color))`

Reply:
I don't think it would work when passing colormaping numbers. `to_color` is noop on `Number` types if I understand. Converting from a Number to RGBA would throw an error. The conversion happens later in `poly!` no ?

Reply:
Ah of course... But should it be always numbers then? the `eltype(plot[:color].val)` makes me nervous :D

Reply:
I actually think it's fine to use the eltype of the original vector (you are basically taking a subset of it, so it is guaranteed to work), but you shouldn't use `plot[:color].val`. `plot[:color]` should be one of the argument of the big `lift` function on top, so that when the color changes, things are updated.

Reply:
Ups, in retrospect I agree with Simon. Even though `eltype` is completely correct in the vector case (as we are taking a subset of the vector, so `eltype` cannot widen), it is not really correct in the scalar case. Here one just get a bit lucky that `eltype` is defined for scalars and returns mostly `Any`. `T = color isa AbstractVector ? eltype(color) : typeof(color)` is the correct type IMO.

Reply:
Ha I was baited because `eltype(1) === Int64`. Changed in https://github.com/JuliaPlots/Makie.jl/pull/1032/commits/1b36cee824a885902dc2ee1bfea5a78caa6d2d20